python/ray/data/dataset.py

```diff
@@ -613,6 +613,34 @@ def do_shuffle(block_list, clear_input_blocks: bool, block_udf, remote_args):
         )
         return Dataset(plan, self._epoch, self._lazy)
 
+    def random_sample(self, number: int, *, seed: Optional[int] = None) -> List[Any]:
+        """Randomly samples N elements from the dataset.
+
+        Examples:
+            >>> import ray
+            >>> ds = ray.data.range(100) # doctest: +SKIP
+            >>> ds.random_sample(5) # doctest: +SKIP
+            >>> # Sample this dataset with a fixed random seed.
+            >>> ds.random_sample(5, seed=12345) # doctest: +SKIP
+
+
+        Args:
+            number: The number of elements to sample from the dataset.
+
+            seed: Seeds the python random pRNG generator.
+
+        Returns:
+            N elements from the shuffled dataset.
+        """
+        import random
+
+        idx = random.randint(0, self.num_blocks())
+        if idx + number >= self.num_blocks():
+            idx = self.num_blocks() - number
+        spliced = self.split_at_indices([idx, idx + number])
```

Comment:
This wouldn't be fully random, since the selected range is contiguous and could be correlated. How about instead generating a number of items to fetch from [0, num_rows], and then gathering them from the corresponding blocks?

Another possibly simpler way is to calculate how many samples are needed per block/batch, and then using `ds.map_batches(sample_n, batch_size=None)` to generate N samples per block. We can collect the samples locally and truncate it to the exact number required.

Reply:
Looks like I overlooked that since I tested with a `ds = ray.data.range(100)`. Is there any way to get a better dataset to test this out?

Reply:
Alternatively, is calling `random_shuffle()` on the dataset and `take`ing N samples a viable solution? Could be inefficient, though.

Reply:
I would try ray.data.range but set the parallelism to something like 5, to simulate more realistic partitioning.

Yeah-- sample usually you want to be much faster.

Reply:
Thanks for the data generator tip! Working on it now.